agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/AgentSpanExporter.java

```diff
@@ -78,4 +96,38 @@ public CompletableResultCode flush() {
   public CompletableResultCode shutdown() {
     return CompletableResultCode.ofSuccess();
   }
+
+  private void exportEvents(SpanData span, TelemetryItem telemetryItem) {
+    for (EventData event : span.getEvents()) {
+      String instrumentationScopeName = span.getInstrumentationScopeInfo().getName();
+      if (eventSuppressor.test(event, instrumentationScopeName)) {
+        continue;
+      }
+      event.getAttributes().forEach((k, v) -> logger.debug("event.attributes: {}:{}", k, v));
+      if (event.getAttributes().get(SemanticAttributes.EXCEPTION_TYPE) != null
+          || event.getAttributes().get(SemanticAttributes.EXCEPTION_MESSAGE) != null) {
+        SpanContext parentSpanContext = span.getParentSpanContext();
+        // Application Insights expects exception records to be "top-level" exceptions
+        // not just any exception that bubbles up
+        if (!parentSpanContext.isValid() || parentSpanContext.isRemote()) {
+          // TODO (trask) map OpenTelemetry exception to Application Insights exception better
+          String stacktrace = event.getAttributes().get(SemanticAttributes.EXCEPTION_STACKTRACE);
+          if (stacktrace != null) {
+            Double samplingPercentage =
+                exceptionSamplingOverrides.getOverridePercentage(event.getAttributes());
+            if (samplingPercentage != null
```

Comment:
Could we avoid these 4 nested statements for readability?

Reply:
this is old code, each condition created a new local variable which is used in the next nested condition.
what will you recommend?